test/jdk/java/awt/List/KeyEventsTest/KeyEventsTest.java

```diff
@@ -66,16 +65,14 @@ public static void main(final String[] args) throws Exception {
         r = new Robot();
         KeyEventsTest app = new KeyEventsTest();
         try {
-            EventQueue.invokeAndWait(app::initAndShowGui);
+            app.initAndShowGui();
```

Comment:
>At the same time, I find it weird that repaint paints directly in its code and then posts paint event.

This is one of the implementation details of the AWT, the "native" component should be painted before the paint event will proceed on EDT, and even if EDT is blocked the "native" part should be painted. Since the XAWT is implemented via Swing which is not thread-safe we should use some additional synchronization. On macOS we have a special lock for that https://github.com/openjdk/jdk/blob/559fc711e03cf0086bea399ffb40cf294cbbb6e1/src/java.desktop/macosx/classes/sun/lwawt/LWListPeer.java#L87 and https://github.com/openjdk/jdk/blob/559fc711e03cf0086bea399ffb40cf294cbbb6e1/src/java.desktop/macosx/classes/sun/lwawt/LWComponentPeer.java#L361 but on XAWT it is quite "cumbersome" since it was implemented a long time ago.

Reply:
And the NPE is triggered for List.select only because it is one method in the test that uses peer w/o synchronization.
https://github.com/openjdk/jdk/blob/77468c284c068f921da543edd28333911e915b61/src/java.desktop/share/classes/java/awt/List.java#L588